tests/unit_tests/sub_classing_test.py

```diff
@@ -0,0 +1,114 @@
+import unittest
```

Comment:
This isn't just a "sub classing test", lets rename it to detection sub classing test.

Reply:
Changed in b6e26294dec0e4bd447938aeb297500abbedc0ae